src/server.ts

```diff
@@ -74,29 +78,27 @@ export class RecordReplayServer {
         return;
       }
 
-      if (
-        this.preventConditionalRequests &&
-        (req.method === "GET" || req.method === "HEAD")
-      ) {
-        // Headers are always coming in as lowercase.
-        delete req.headers["if-modified-since"];
-        delete req.headers["if-none-match"];
-      }
```

Comment:
Move this logic into the new `rewriteRequest` helper method that includes other potential request modifications.